JS/ViewControllers/libraryChooseMediaLanguageViewController.js

```diff
@@ -0,0 +1,118 @@
+        this.availableLanguages;
+    }
+
+    renderHtml(html) {
+        const htmlView = `
+            <section id="LibraryChooseMediaLanguageViewController" class="container">
+                <div class="row row-100 m12">
+                    <div class="col s12">
+                        <h1 class="center">Choose media language</h1>
+                    </div>
+                </div>
+                <div class="row row-25 m12">
+                    <form action="#">
+                        <select id="select-language" class="browser-default">
+                            <option value="none" selected></option>
+                        </select>
+                    </form>
+                </div>
+                <p id="demo"></p>
+                <div class="row row-50">
+                    <div class="col s12 center">
+                        <a id="choose-button" class="waves-effect waves-light btn-large disabled">choose</a>
+                    </div>
+                </div>
+                <div class="row row-100">
+                    <div class="col s12">
+                        <a class="btn-small right" href="index.html">Back to my menu</a>
+                    </div>
+                </div>
+            </section>
+        `;
+        super.renderHtml(htmlView);
+      }
+
+      setupProperties() {
+        this.languagePicker = $('#select-language');
+        this.chooseButton = $('#choose-button');
+        this.getAvailableLanguage();
+      }
+
+      setupEventListeners() {
+        this.languageValueChanged = this.languageValueChanged.bind(this);
+        this.chooseButtonClicked = this.chooseButtonClicked.bind(this);
+
+        this.languagePicker.change(this.languageValueChanged);
+        this.chooseButton.on('click', this.chooseButtonClicked);
+      }
+
+      presentNextController() {
+        const librarySelectMediaLanguageViewController = new LibrarySelectMediaLanguageViewController();
+        librarySelectMediaLanguageViewController.language = this.language;
+        librarySelectMediaLanguageViewController.shortLanguage = this.shortLanguage;
+        this.navigationController.present(librarySelectMediaLanguageViewController);
+      }
+
+      languageValueChanged(){
+        var e = document.getElementById("select-language");
+        this.language = e.options[e.selectedIndex].text;
+        this.shortLanguage = e.options[e.selectedIndex].value;
+        this.setupChooseButton();
+      }
+
+      setupChooseButton() {
+          const shouldEnableSelectButton = this.language !== "";
+          if (shouldEnableSelectButton) {
+              this.chooseButton.removeClass("disabled");
+          } else {
+              this.chooseButton.addClass("disabled");
+          }
+      }
+
+      chooseButtonClicked() {
+          this.presentNextController();
+      }
+
+      // Private Methods
+
+      // get available labguage from online library
+      getAvailableLanguage(){
+        var tmp = "Null";
```

Comment:
Aky ma toto zmysel? Po prve, preco je v tom nejaky string Null? Je to nejaky javascript practice, nasiel si to na nete? Dalej preco rovno do `this.availableLanguages` v success funkcii nepriradis tie data rovno a robis to cez tmp? a keby si do toho chcel priradit nejaky stav ktory neni ok, tak tam mas predsa na to error funkciu

Reply:
No s tymto som mal velky problem ked som to robil. Netusim presne preco, ale ked sa to nerobilo cez tu pomocnu premennu **var tmp** tak mi neslo tie data ulozit a pracovat s nimi. Pomahal mi pri tom aj Mato a takto to jedine spravne fungovalo.